pkg/sharding/clients.go

```diff
@@ -0,0 +1,53 @@
+package sharding
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/genericcontrolplane"
+)
+
+type IdentifiedConfig struct {
+	Identifier string
+	Config     *rest.Config
+}
+
+type ClientLoader struct {
+	*sync.RWMutex
+	Clients map[string]*rest.Config
```

Comment:
I think if you're embedding a mutex in this struct and using it to guard the clients map, you wouldn't want to export Clients to users, as that would allow them to ignore the mutex.

I'd do Clients -> clients, and optionally decide if you want ClientLoader.Lock/Unlock to be a thing, or if you'd rather present a finer-grained API to consumers (e.g. GetClient or whatever you'd like to call it).

Reply:
@stevekuznetsov we can merge this for now, but can you add a FIXME here?